portal_tables/sync_publications.py

```diff
@@ -34,81 +37,109 @@ def add_missing_info(pubs, grants):
     pubs.loc[:, 'Link'] = [
         "".join(["[PMID:", str(pmid), "](", url, ")"])
         for pmid, url
-        in zip(pubs.pubmedId, pubs.pubmedUrl)
+        in zip(pubs['Pubmed Id'], pubs['Pubmed Url'])
     ]
-    pubs['grantName'] = ""
-    for i, row in pubs.iterrows():
-        grant_names = []
-        for g in row['publicationGrantNumber']:
-            grant_names.append(
-                grants[grants.grantNumber == g]['grantName'].values[0])
-        pubs.at[i, 'grantName'] = grant_names
+    
+    pattern = re.compile("(\')([\s\w/-]+)(\')")
+    
+    for col in new_cols: 
+        pubs[col] = ""
+        for row in pubs.itertuples():
+            i = row[0]
+            n = row[4].split(",")
+            extracted = []
+            for g in n:
+                if len(grants[grants.grantNumber == g][col].values) > 0:
+                    values = str(grants[grants.grantNumber == g][col].values[0])
+                    
+                    if col == 'grantName':
+                        extracted.append(values)
+                    
+                    else:
+                        matches = pattern.findall(values)
+                        for m in matches:
+                            extracted.append(m[1])
+
+                else:
+                    print(f"No match found for grant number: {g}")
+                    continue
+            
+            clean_values = ', '.join(list((dict.fromkeys(extracted))))
```

Comment:
We may need to add an additional check here before joining the list as a string.  IIRC, `theme` and `consortium` are STRINGLIST in the Merged table, so we would want to save `clean_values` as a list rather than a string.

Reply:
Oh!  Perhaps a check is not needed after all, as `grantName` is also a STRINGLIST.  In that case, we should remove the `join()`